avalon/pipeline.py

```diff
@@ -248,6 +248,56 @@ def process(self, session, **kwargs):
         pass
 
 
+class InventoryAction(object):
+    """A custom action for the scene inventory tool
+
+    If registered the action will be visible in the Right Mouse Button menu
+    under the submenu "Actions".
+
+    """
+
+    label = None
+    icon = None
+    color = None
+    order = 0
+
+    @staticmethod
+    def is_compatible(container):
+        """Override function in a custom class
+
+        This method is specifically used to ensure the action can operate on
+        the container.
+
+        Args:
+            container: dict
+
+        Returns:
+            bool
+        """
+        return True
+
+    def process(self, containers):
+        """Override function in a custom class
+
+        This method will receive all containers even those which are
+        incompatible. It is advised to create a small filter along the lines
+        of this example:
+
+        valid_containers = [c for c in containers if "myKey" in c]
```

Comment:
I don't understand this example. ;)

Shouldn't this example just use the `is_compatible` method?
```python
containers = filter(self.is_compatible, containers) 
```

Reply:
You are right